regulation-worker/internal/delete/api/api.go

```diff
@@ -106,22 +105,28 @@ func (api *APIManager) Delete(ctx context.Context, job model.Job, destination mo
 		return model.JobStatusFailed
 	}
 	jobStatus := getJobStatus(resp.StatusCode, jobResp)
-	pkgLogger.Debugf("[%v] JobStatus for %v: %v", destination.Name, destination.DestinationID, jobStatus)
+	pkgLogger.Debugf("[%v] Job: %v, JobStatus: %v", destination.Name, job.ID, jobStatus)
 
-	if isOAuthEnabled && isTokenExpired(jobResp) {
+	if isOAuthEnabled && isTokenExpired(jobResp) && currentOauthRetryAttempt < api.MaxOAuthRefreshRetryAttempts {
```

Comment:
what the `jobStatus` will be when `isOAuthEnabled && isTokenExpired && currentOauthRetryAttempt >= api.MaxOAuthRefreshRetryAttempts`?

Reply:
Whatever is calculated as part of `getJobStatus()` that we got in the last retry(when `currentOauthRetry = api.MaxOAuthRefreshRetryAttempts - 1`)

Reply:
There is already a retry mechanism such that regulation-manager would provide the already failed jobs to regulation-worker to send it to destinaiton, so that way I feel it should be ok. Let me know if you think otherwise

Reply:
I will try to rephrase the question:
What will the transformer's **HTTP response code** be whenever it returns in its response body at least 1 entry with `authErrorCategory=REFRESH_TOKEN`? Is it deterministic or not?

Reply:
Whenever atleast one entry of `authErrorCategory = REFRESH_TOKEN` is present in response body, we will return a status-code of `500` as transformer http status-code as of today.

Reply:
When you say deterministic, are you asking about how do we identify perpetual refresh token failures, I mean the scenario wherein even after multiple retries(when `api.MaxOAuthRefreshRetryAttempts > 1`), we are still getting `refresh_token` error from transformer or may be it is something else ?

Reply:
Just trying to make sure that the jobStatus will be `model.JobStatusFailed` if the token refresh fails or we have reached to maximum refresh attempts :)